pkg/core/owner_references.go

```diff
@@ -123,13 +123,25 @@ func getOrphans(existing []Object, current []configObject) []Object {
 func getOwnerReference(obj podController) metav1.OwnerReference {
 	t := true
 	f := false
-	return metav1.OwnerReference{
-		APIVersion:         "apps/v1",
-		Kind:               "Deployment",
-		Name:               obj.GetName(),
-		UID:                obj.GetUID(),
-		BlockOwnerDeletion: &t,
-		Controller:         &f,
+	switch kindOf(obj) {
+	case "Deployment":
+		return metav1.OwnerReference{
+			APIVersion:         "apps/v1",
+			Kind:               "Deployment",
```

Comment:
Any reason not to just substitute the string here for `kindOf(obj)`, then this switch would be unnecessary right?
```suggestion
			Kind:               kindOf(obj),
```